EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py

```diff
@@ -3,18 +3,20 @@
 from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
 from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
 from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+from Configuration.Eras.Modifier_ctpps_2022_cff import ctpps_2022
 
 from EventFilter.CTPPSRawToDigi.totemVFATRawToDigi_cfi import totemVFATRawToDigi
 
 ctppsDiamondRawToDigi = totemVFATRawToDigi.clone(
     subSystem = 'TimingDiamond',
-    fedIds = [579, 581, 582, 583], #as declared in DataFormats/FEDRawData/interface/FEDNumbering.h
+    fedIds = [579, 581, 582, 583, 588, 589], #as declared in DataFormats/FEDRawData/interface/FEDNumbering.h
```

Comment:
Is the expectation that these values would be used in the (prompt) reconstruction of 2023 data?

Reply:
Yes, for 2023 we expect to use two new FEDs, for all other datasets we have more limited list, see code below.

Reply:
> Thanks. As far as I can tell based on how things are set up currently in this PR, these values get ignored. That happens because there is only one `Run3` Era (ModifierChain), and that includes the `ctpps_2022` Modifier, that overrides the values to `[579, 581, 582, 583]` below.

I believed that `ctpps_2022` modifier was included in Run3 era, but valid only for runs from 2022.
What you suggests leads indeed to the fact that the fedIDs will be ignored for 2023 runs.

> It seems to me this PR would need to "split" the `Run3` Era into year-specific Eras, as was done with `Run2` eras. I believe the setup would be something along (similar to 12_6_X-specific #40756, but more complicated because of creating a baseline for long-term setup)
> 
> * add `ctpps_2023` Modifier, and use that to set the 2023-specific values

That make sense and it seems a straightforward modification.

> * set up the `Run3_2023` ModifierChain as `Run3.copyAndExclude([ctpps_2022], ctpps_2023)`

Where this change should be applied ? In `Configuration/Eras/python/Modifier_ctpps_2023_cff.py` ?

>   * as far as I can tell, the `ctpps_*` Modifiers are year-specific and should not carry over
> * add `ctpps_2023` in necessary places that use `ctpps_2022` so that those customizations apply to 2023 as well

OK

> * update `Phase2` Era to start from `Run3_2023` instead of `Run3`,

How this should be modified ?

> and exclude `ctpps_2023` instead of `ctpps_2022`

Do you mean modification in 
https://github.com/cms-sw/cmssw/blob/master/Configuration/Eras/python/Era_Phase2_cff.py#L21
?

> * update all 2023 and 2024 runTheMatrix workflows to use `Run3_2023` era instead of `Run3`

That seems beyond the scope of this PR. Maybe another PR should be created which adjust Eras for PPS (I have no idea how to validate such a change) and then rebase this on top of the other.

>   * it is unclear to me which of the `Run3_dd4hep`, `Run3_DDD`, `Run3_FastSim`, `Run3_noMkFit`, `Run3_pp_on_PbPb_approxSiStripClusters`, `Run3_pp_on_PbPb` would need to be copied/moved to `Run3_2023`

Me too

> Eventually it would probably be good to rename `Run3` era to `Run3_2022`, and propagate that change to all runTheMatrix workflows etc.

That seems a lot of changes

Reply:
That indeed seems like a lot of changes, but I'm not immediately seeing an easy way around it.  @perrotta @rappoccio any suggestions how to proceed?  Should we merge this and then someone (could be me) makes new PR to split the Run 3 era in two?

Reply:
> > * set up the `Run3_2023` ModifierChain as `Run3.copyAndExclude([ctpps_2022], ctpps_2023)`
> 
> Where this change should be applied ? In `Configuration/Eras/python/Modifier_ctpps_2023_cff.py` ?

There would need to be a new file, `Configuration/Eras/python/Era_Run3_2023_cff.py`.

> > * update `Phase2` Era to start from `Run3_2023` instead of `Run3`,
> 
> How this should be modified ?
> 
> > and exclude `ctpps_2023` instead of `ctpps_2022`
> 
> Do you mean modification in https://github.com/cms-sw/cmssw/blob/master/Configuration/Eras/python/Era_Phase2_cff.py#L21 ?

Yes.

> That seems beyond the scope of this PR.
> ...
> That seems a lot of changes

I agree with both of these.  I think it would be cleanest if someone(TM) would do the technical "branching" of `Run3` Era in a separate PR, and this PR would then build on top of that (but there may be practical considerations favoring some more "approximate" way forward).

Reply:
There is also another aspect of the new eras. Introducing them (in particular ctpps 2023 era) would mean an extensive testing of the PPS reconstruction. We need to be sure that nothing is broken with 2022 -> 2023 switch.
I am afraid there is not enough time to do it with 13_0_X release.

On the other hand we can implement solution described in:
https://github.com/cms-sw/cmssw/pull/40763#issuecomment-1431041711
That doesn't require new ctpps era for 2023.